From ff17d94b7269be679b1cd5887aa4b94f73871913 Mon Sep 17 00:00:00 2001
From: Herton Ronaldo Krzesinski <herton@mandriva.com.br>
Date: Tue, 23 Feb 2010 13:59:20 -0300
Subject: [PATCH 1/8] _nl_load_locale() incorrectly handles mmap() failures

_nl_load_locale() incorrectly handles mmap() failures
Add BZ number.

(cherry-picks from 8b2f25c23374fe79645499b8095f0d2f6eb24f71,
8549abcb9ca4d7857cc457aaa17c2073522804f4)
---
 ChangeLog           |    6 ++++++
 locale/loadlocale.c |    1 +
 2 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 2223015..2b7b746 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,9 @@
+2010-01-22  Ulrich Drepper  <drepper@redhat.com>
+
+	[BZ #11200]
+	* locale/loadlocale.c (_nl_load_locale): Fix recognition of genuine
+	mmap resource problem.  Patch by Joe Landers <jlanders@vmware.com>.
+
 2010-01-22  Jim Meyering  <jim@meyering.net>
 
 	[BZ #11193]
diff --git a/locale/loadlocale.c b/locale/loadlocale.c
index 467dff1..b91941e 100644
--- a/locale/loadlocale.c
+++ b/locale/loadlocale.c
@@ -224,6 +224,7 @@ _nl_load_locale (struct loaded_l10nfile *file, int category)
 		     PROT_READ, MAP_FILE|MAP_COPY, fd, 0);
   if (__builtin_expect (filedata == MAP_FAILED, 0))
     {
+      filedata = NULL;
       if (__builtin_expect (errno, ENOSYS) == ENOSYS)
 	{
 #endif	/* _POSIX_MAPPED_FILES */
-- 
1.7.0

